### PR TITLE
Remote api startRecord() and stopRecord() can detect the correct activity itself.

### PR DESCRIPTION
**Pull request by resli**
- `startRecord($apiKey, "", "")` restarts the last stopped activity,
- `stopRecord($apiKey, "")` stops the current running activity.

This is because the bde-terminal does not knows the activity to start/stop. It only wants to start the last run or stop the current one.

### CODE REVIEW

File: libraries/Kimai/Remote/Api.php [hunk 1]
```diff
@@ -244,13 +244,27 @@ protected function getSuccessResult(array $items, $total = 0)
...
      */
     public function startRecord($apiKey, $projectId, $activityId)
     {
         if (!$this->init($apiKey, 'startRecord')) {
             return $this->getAuthErrorResult();
         }
 
+        // check for empty params
+        if ($projectId=="" ||
+            $activityId==""
```
Comment (simonschaufi) [inline libraries/Kimai/Remote/Api.php]: We use PSR 2 code style. Please format your code accordingly.

**Comment by resli** [comment]
I reformatted the code. I hope it's ok now.

File: libraries/Kimai/Remote/Api.php [hunk 2]
```diff
@@ -244,13 +244,25 @@ protected function getSuccessResult(array $items, $total = 0)
...
+     * If $projectId or $activityId is empty the last activity will be restarted.
      */
     public function startRecord($apiKey, $projectId, $activityId)
     {
         if (!$this->init($apiKey, 'startRecord')) {
             return $this->getAuthErrorResult();
         }
 
+        // check for empty params
+        if (($projectId=="") || ($activityId=="")) {
```
Comment (simonschaufi) [inline libraries/Kimai/Remote/Api.php]: I wonder why styleci is not complaining but there should be spaces around the equal signs and you can also remove the inner brackets. Please use single quotes also.

File: libraries/Kimai/Remote/Api.php [hunk 3]
```diff
@@ -244,13 +244,25 @@ protected function getSuccessResult(array $items, $total = 0)
      * @param integer $projectId
      * @param integer $activityId
      * @return array
+     *
+     * If $projectId or $activityId is empty the last activity will be restarted.
```
Comment (simonschaufi) [inline libraries/Kimai/Remote/Api.php]: This comment must be above the params

File: libraries/Kimai/Remote/Api.php [hunk 4]
```diff
@@ -282,13 +294,34 @@ public function startRecord($apiKey, $projectId, $activityId)
      * @param string $apiKey
      * @param integer $entryId
      * @return boolean
+     *
+     * If $entryId is empty the current activity will be stopped.
```
Comment (simonschaufi) [inline libraries/Kimai/Remote/Api.php]: Dito

File: libraries/Kimai/Remote/Api.php [hunk 5]
```diff
@@ -282,13 +294,34 @@ public function startRecord($apiKey, $projectId, $activityId)
...
      */
     public function stopRecord($apiKey, $entryId)
     {
         if (!$this->init($apiKey, 'stopRecord')) {
             return $this->getAuthErrorResult();
         }
 
+        // check for empty params
+        if ($entryId=="") {
```
Comment (simonschaufi) [inline libraries/Kimai/Remote/Api.php]: Here also

**Comment by resli** [comment]
I hope everything is ok now. Sorry for the inconvenience.

File: libraries/Kimai/Remote/Api.php [hunk 6]
```diff
@@ -251,6 +252,16 @@ public function startRecord($apiKey, $projectId, $activityId)
             return $this->getAuthErrorResult();
         }
 
+        // check for empty params
+        if ($projectId == '' || $activityId == '') {
```
Comment (kevinpapst) [inline libraries/Kimai/Remote/Api.php]: From my POV this should be changed to 
`if ($projectId == '' && $activityId == '') {`
If only one of the values is empty, this is an invalid requestby the next check for backend->isValidX()

If both are empty we could assume, that the last task should be re-started.

**Comment by resli** [comment]
I agree with you and modified the commit.

**Comment by kevinpapst** [comment]
Thanks @resli 👍

**Pull request merged**